src/redismodule.rs

```diff
@@ -126,20 +116,42 @@ impl RedisString {
         len == 0
     }
 
-    pub fn try_as_str(&self) -> Result<&str, Utf8Error> {
+    pub fn try_as_str(&self) -> Result<&str, RedisError> {
         Self::from_ptr(self.inner)
+            .map_err(|_e| RedisError::String(format!("Couldn't parse as UTF-8 string")))
     }
 
     /// Performs lossy conversion of a `RedisString` into an owned `String. This conversion
     /// will replace any invalid UTF-8 sequences with U+FFFD REPLACEMENT CHARACTER, which
     /// looks like this: �.
-    pub fn into_string_lossy(self) -> String {
+    pub fn into_string_lossy(&self) -> String {
         let mut len: libc::size_t = 0;
         let bytes = unsafe { raw::RedisModule_StringPtrLen.unwrap()(self.inner, &mut len) };
         let bytes = unsafe { slice::from_raw_parts(bytes as *const u8, len) };
         String::from_utf8_lossy(bytes).into_owned()
     }
 
+    pub fn parse_unsigned_integer(&self) -> Result<u64, RedisError> {
+        let val = self.parse_integer()?;
+        u64::try_from(val).map_err(|_e| RedisError::String(format!("Couldn't parse as integer")))
+    }
+
+    pub fn parse_integer(&self) -> Result<i64, RedisError> {
+        let mut val: i64 = 0;
+        match raw::string_to_longlong(self.inner, &mut val) {
+            raw::Status::Ok => Ok(val),
+            raw::Status::Err => Err(RedisError::String(format!("Couldn't parse as integer"))),
```

Comment:
No need for the `format!`:
```suggestion
            raw::Status::Err => Err(RedisError::Str("Couldn't parse as integer")),
```